app/models/concept-engagement.ts

```diff
@@ -10,10 +10,31 @@ export default class ConceptEngagementModel extends Model {
   @attr('date') declare lastActivityAt: Date;
   @attr('date') declare startedAt: Date;
 
-  get remainingBlocksCount() {
-    const allBlocks = this.concept.parsedBlocks;
-    const completedBlocksCount = Math.round((this.currentProgressPercentage / 100) * allBlocks.length);
+  get completedBlockGroupsCount() {
+    const completedBlocksCount = Math.round((this.currentProgressPercentage / 100) * this.totalBlocksCount);
 
-    return allBlocks.length - completedBlocksCount;
+    return this.convertBlockProgressIntoBlockGroupProgress(this.concept.allBlockGroups, completedBlocksCount);
+  }
+
+  get totalBlockGroupsCount() {
+    return this.concept.allBlockGroups.length;
+  }
+
+  get totalBlocksCount() {
+    return this.concept.parsedBlocks.length;
+  }
+
+  convertBlockProgressIntoBlockGroupProgress(allBlockGroups: BlockGroup[], completedBlocksCount: number) {
```

Comment:
This function isn't re-used anywhere, so the logic can remain within `completedBlockGroupsCount` directly. 

Just add `completedBlocksCount` as one method, and then use that within `completedBlockGroupsCount` using the specified logic.